src/wallet/wallet.cpp

```diff
@@ -6750,6 +6756,132 @@ bool CWallet::InitLoadWallet(const CChainParams& params, bool clearWitnessCaches
+                auto fingerprintPos = line.find("fingerprint=");
+                if (fingerprintPos != std::string::npos) {
+                    if (!(recoveryPhrase.has_value() && lineNumber == (lastUsedLine + 1))) {
+                        return UIError(_("The 'fingerprint=' line must immediately follow the last used line."));
+                    }
+                    lastUsedLine++;
+
+                    auto seedFpStr = line.substr(fingerprintPos + 12, std::string::npos);
+                    boost::trim(seedFpStr);
+                    uint256 seedFp;
+                    seedFp.SetHex(seedFpStr.c_str());
+                    fingerprint = seedFp;
+                }
+
+                // Syntax:
+                //     num_accounts=INT
+                //
+                // Constrained to immediately follow fingerprint= (if present).
+                //
+                // If this line is not present, treat it as num_accounts=1.
+                // If num_accounts=0, treat it as such (no automatic account recovery).
+                auto numAccountsPos = line.find("num_accounts=");
+                if (numAccountsPos != std::string::npos) {
+                    if (!(recoveryPhrase.has_value() && lineNumber == (lastUsedLine + 1))) {
+                        return UIError(_("The 'num_accounts=' line must immediately follow the last used line."));
+                    }
+                    lastUsedLine++;
+
+                    auto numAccountsStr = line.substr(numAccountsPos + 13, std::string::npos);
+                    boost::trim(numAccountsStr);
+                    numAccounts = static_cast<uint32_t>(std::stoul(numAccountsStr.c_str()));
+                }
+            }
+        }
+
+        if (!recoveryPhrase.has_value()) {
+            return UIError(_("Unable to parse mnemonic phrase from recovery file."));
+        }
+
+        auto seed = MnemonicSeed::ForPhrase(
+            language.value_or(MnemonicSeed::LanguageFromName("English").value()),
+            recoveryPhrase.value());
+
+        if (!seed.has_value()) {
+            return UIError(_("Unable to generate a valid seed from the given recovery phrase and language."));
+        }
+
+        if (!language.has_value()) {
+            return UIError(_("Incorrect language provided for the given recovery phrase. Check that the phrase is correct, and add a language=LANGUAGE_NAME line if the recovery phrase language is not English."));
+        }
+
+        // Check against fingerprint, if available
+        if (fingerprint.has_value()) {
+            assert(seed.value().Fingerprint() == fingerprint.value());
+        }
+
+        walletInstance->InitWithSeed(seed.value());
+        for (uint32_t acctId = 0; acctId < numAccounts; acctId++) {
+            walletInstance->GenerateNewUnifiedSpendingKey();
+            // walletInstance->GenerateUnifiedAddress(acctId, CWallet::DefaultReceiverTypes(chainActive.Height()), std::nullopt);
```

Comment:
Was something intended to be done with this commented-out line? Does `GenerateNewUnifiedSpendingKey` produce the required default address internally?

In any case, I'm not sure that this is sufficient, because the wallet may have generated an arbitrary number of transparent addresses corresponding to successive diversifiers, so we need to generate some number of (say 20) diversified addresses, then, as we're scanning, if we observe transparent funds received to one of these addresses, we should generate additional diversified addresses until we have a buffer of 20 unused transparent addresses again.